sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RequireFailForTryCatchInJunitCheck.java

```diff
@@ -88,9 +99,13 @@
      */
     private boolean importTest;
     /**
-     * {@code true} if the junit assert is imported.
+     * {@code true} if the junit 4 assert is imported.
+     */
+    private boolean importJunit4Assert;
+    /**
+     * {@code true} if the junit 5 assertions is imported.
      */
-    private boolean importAssert;
+    private boolean importJunit5Assertions;
```

Comment:
Why do we need to identify junit 4 support versus 5? If someone imports an assertion, then it is slightly hard to use another assert without using the fully qualified class.

Most common cases people will use one junit version or the other. I did happen to mix and match 4 and 5 in main repo since my eclipse doesn't support junit 5 and I was able to run junit 4 Test with junit 5 assertions.

Reply:
We can clone this check to verify Junit5 assertions only. Users will have to specify which check they want to use in their config. @romani what do you think?

Reply:
> We can clone this check to verify Junit5 assertions only. Users will have to specify which check they want to use in their config.

This doesn't specify why it is needed. Why do we need to know exact junit version user is using for validation?

Reply:
> Why do we need to know exact junit version user is using for validation?

We do not need it. We need to know which assertion class is imported: `Assert` from Junit 4 or `Assertions` from Junit 5. To avoid false positives like this:
```
import org.junit.Assert;
import org.example.Assertions;

// ....
    try {
       foo();
       Assertions.fail("I'm a wrong fail method"); 
    }
```
For static import of the method `fail` and for annotation `Test` we do not need to know which framework is in use.

Reply:
Thank you, I understand now. I am fine to keep it all in 1.

Reply:
@pbludov , @rnveach , ideally it would be better to make it configurable from user properties to user himself define classes to consider as assert/fail. Junit3/4 was with us too long, but there bunch of other tools and custom `fail` methods, and this Check only care that some recognizable method is called.